staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -591,3 +595,7 @@ func patchToUpdateOptions(po *metav1.PatchOptions) (*metav1.UpdateOptions, error
 	err = json.Unmarshal(b, &uo)
 	return &uo, err
 }
+
+func stripUserAgent(userAgent string) string {
```

Comment:
Should we default something here, and maybe rename the function? And maybe make it a method of the patcher so that we don't have to pass it the userAgent every time?

Reply:
I was thinking about different ways to do this as well.
My decission for now was to store the userAgent in patcher and keep it clean because it is a more generic value than something like "managerPrefix". Open to change that though.

What do you mean by not having to pass patcher the userAgent. As far as I can see, we need it in patcher as it is only available up when patcher gets initialized.

Reply:
Ah, nevermind reading it a second time solved my question about the last part...